test/test-fft-native.cpp

```diff
@@ -0,0 +1,226 @@
+// Copyright (C) 2020-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+
+#include <complex>
+
+#include "hexl/fft/fft-native.hpp"
+#include "hexl/fft/fft.hpp"
+#include "hexl/logging/logging.hpp"
+#include "hexl/util/defines.hpp"
+#include "test-util.hpp"
+#include "util/cpu-features.hpp"
+#include "util/util-internal.hpp"
+
+namespace intel {
+namespace hexl {
+
+TEST(FFT, ForwardInverseFFTNative) {
+  FFT fft(64, nullptr);
+  AlignedVector64<std::complex<double>> root_powers =
+      fft.GetComplexRootsOfUnity();
+  AlignedVector64<std::complex<double>> inv_root_powers =
+      fft.GetInvComplexRootsOfUnity();
+
+  {  // Single Unscaled
+    const uint64_t n = 64;
+    const double data_bound = (1 << 30);
+    AlignedVector64<std::complex<double>> operand(n);
+    AlignedVector64<std::complex<double>> result(n);
+
+    operand[0] = std::complex<double>(
+        GenerateInsecureUniformRealRandomValue(0, data_bound),
+        GenerateInsecureUniformRealRandomValue(0, data_bound));
+
+    Forward_FFT_ToBitReverseRadix2(result.data(), operand.data(),
+                                   root_powers.data(), n);
+
+    for (size_t i = 0; i < n; ++i) {
+      double tmp = abs(operand[0].real() - result[i].real());
+      ASSERT_TRUE(tmp < 0.5);
+      tmp = abs(operand[0].imag() - result[i].imag());
+      ASSERT_TRUE(tmp < 0.5);
```

Comment:
Would this not work? I would also use EXPECTs here to not shadow the other result. https://github.com/google/googletest/blob/main/docs/reference/assertions.md#expect_near-expect_near
```suggestion
      EXPECT_NEAR(operand[0].real(), result[i].real()), 0.5)
      EXPECT_NEAR(operand[0].imag(), result[i].imag(), 0.5);
```